CMakeLists.txt

```diff
@@ -7,9 +6,22 @@ ENDIF()
 
 IF(NOT KOKKOS_HAS_TRILINOS)
   CMAKE_MINIMUM_REQUIRED(VERSION 2.8.11 FATAL_ERROR)
+  # hack in openmp
+  IF (Kokkos_ENABLE_OpenMP)
+    FIND_PACKAGE (OpenMP REQUIRED)
+    IF (OPENMP_FOUND)
+      SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+      SET (CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${OpenMP_C_FLAGS}")
+    ENDIF ()
+  ENDIF (Kokkos_ENABLE_OpenMP)
+  IF (Kokkos_ENABLE_CUDA)
+    SET (CMAKE_CXX_COMPILER "${CMAKE_SOURCE_DIR}/bin/nvcc_wrapper")
+  ENDIF ()
   INCLUDE(cmake/tribits.cmake)
 ENDIF()
 
+SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
```

Comment:
TriBITS automatically deals with support for C++11.  See:

https://github.com/kokkos/kokkos/issues/155#issuecomment-229360446

No reason to re-invent the wheel.